packages/pancake-uikit/src/hooks/useTooltip/useTooltip.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from "react";
+import { Placement, Padding } from "@popperjs/core";
+import { usePopper } from "react-popper";
+import { StyledTooltip, Arrow } from "./StyledTooltip";
+import { TooltipRefs, TriggerType } from "./types";
+
+function isTouchDevice() {
+  return "ontouchstart" in window || navigator.maxTouchPoints > 0 || navigator.msMaxTouchPoints > 0;
+}
+
+const useTooltip = (
+  content: string | React.ReactNode,
+  placement: Placement = "auto",
+  trigger: TriggerType = "hover",
+  arrowPadding?: Padding,
+  tooltipPadding?: Padding,
+  tooltipOffset?: [number | null | undefined, number | null | undefined]
```

Comment:
Because we do not use `strictNullTypes` in Typescript (should we?) specifying `number | null | undefined` is not necessary.

From the Typescript docs
>By default null and undefined are subtypes of all other types. That means you can assign null and undefined to something like number.

> However, when using the --strictNullChecks flag, null and undefined are only assignable to unknown, any and their respective types (the one exception being that undefined is also assignable to void). This helps avoid many common errors. In cases where you want to pass in either a string or null or undefined, you can use the union type string | null | undefined.